dn1/scripts/demo.jl

```diff
@@ -0,0 +1,197 @@
+Indeksi v vektorju `b` ustrezajo vozliščem v istem vrstnem redu,
+kot nastopajo v argumentu `sprem`.
+"""
+function desne_strani(G::AbstractGraph, sprem, koordinate)
+    set = Set(sprem)
+    m = length(sprem)
+    b = zeros(m)
+    for i = 1:m
+        v = sprem[i]
+        for v2 in neighbors(G, v)
+            if !(v2 in set) # dodamo le točke, ki so fiksirane
+                b[i] -= koordinate[v2]
+            end
+        end
+    end
+    return b
+end
+
+"""
+    vloži!(G::AbstractGraph, fix, točke)
+
+Poišči vložitev grafa `G` v prostor s fizikalno metodo. Argument `fix` vsebuje
+vektor vozlišč grafa, ki imajo določene koordinate. Argument `točke` je
+začetna vložitev grafa. Koordinate vozlišč, ki niso pritrjena, bodo nadomeščene
+z novimi koordinatami.
+Metoda ne vrne ničesar, ampak zapiše izračunane koordinate v matriko `točke`.
+"""
+function vloži!(G::AbstractGraph, fix, točke, omega)
+    sprem = setdiff(vertices(G), fix)
+    dim, _ = size(točke)
+    A = matrika(G, sprem)
+    x0 = ones(size(A)[1])
+    it = 0
+    for k = 1:dim
+        b = desne_strani(G, sprem, točke[k, :])
+        x = sor(A, b, x0, omega)#x = cg(-A, -b) # matrika A je negativno definitna, mejbi probi z -
+        #println(x)
+        #println(b)
+        #println(length(sprem))
+        #println(length(x))
+        točke[k, sprem] = x[1]
+        x0 = x[1]
+        it += x[2]
+    end
+    return it
+end
+
+
+G = krožna_lestev(10)
+tocke = zeros(2, nv(G))
+fix = [1, 4, 7]
+tocke[:, 1] = [0.0, 0.0]
+tocke[:, 4] = [1.0, 0.0]
+tocke[:, 7] = [0.5, 1.0]
+
+bestit = 10000
+bestomega = 0
+its = []
+for omega in 0.1:0.1:2
+    println(omega)
+    global bestit
+    global bestomega
+    tocke = zeros(2, nv(G))
+    fix = [1, 4, 7]
+    tocke[:, 1] = [0.0, 0.0]
+    tocke[:, 4] = [1.0, 0.0]
+    tocke[:, 7] = [0.5, 1.0]
+    it = vloži!(G, fix, tocke, omega)
+    if it < bestit
+        bestit = it
+        bestomega = omega
+    end
+    push!(its, it)
+end
+
+it = vloži!(G, fix, tocke, 0.5)
+println(it)
+println(bestit)
+println(bestomega)
+#println(tocke)
+
+
+"""
+    narisi_vlozitev(G::AbstractGraph, točke)
+
+funkcija nariše pozicije vozlišč v grafu G z podanimi koordinatami v matriki točke, kjer vsak stolpec predstavlja eno torčko, vrstici pa x in y koordinato
+"""
+function narisi_vlozitev(G::AbstractGraph, točke)
+    x = točke[1, :]
+    y = točke[2, :]
+
+    # Draw edges
+    for e in edges(G)
+        v1 = src(e)
+        v2 = dst(e)
+        plot!([x[v1], x[v2]], [y[v1], y[v2]], c=:gray, lw=1)
+    end
+
+    # Draw vertices
+    scatter!(x, y, c=:red, ms=5, label=false)
+
+    # Display the plot
+    plot!(aspect_ratio=:equal, legend=false)
+end
+
+plot()
+narisi_vlozitev(G, tocke)     
+#' 
+
+#' Za iskanje optimalnega $\omega$ parametra metode SOR lahko pogledamo, koliko iteracij je potrebno za izračun optimalne vložitve na 10 decimalnih mest natančno. Najmanj iteracij je potrebno pri $\omega = 1.4$ in sicer 98 iteracij
```

Comment:
Kako ste prišli do 1.4? Predvidevam, da ste poskusili vse omege med 0 in 1 z eno decimalko: 0, 0.1, 0.2, ...? Napišite to!